### ISSUE TITLE
[Regression] GENERIC_CHART_AXES: numeric x-axis treated as categorical values

### ISSUE THREAD

**Issue by rumbin** [apache/superset#26034]
When using a _numeric_ column as the generic x-axis, the x-axis values aren't linearly distributed along the x-axis but they are evenly distributed, just as if they were categorical values

#### How to reproduce the bug

0. Have the `GENERIC_CHART_AXES` feature flag enabled
1. Create a dataset or a SQLLab query with numerical columns, e.g.

```sql
select
  *
from (values
  (1,1)
  , (2,2)
  , (5,5)
  , (10,10)
) as v(x,y)
```

2. Visualize the query/dataset as bar chart (or line chart)
3. Use `x` as x-axis and, e.g., `max(y)` as y-axis
4. See that the x-axis values are shown with equal spacing instead of the expected linear distribution of the x values

### Expected results

This is what it should look like and what it did correctly in Superset 2.0.1:

![image](https://github.com/apache/superset/assets/1220356/baaa44e9-b397-4f77-bb74-ff545d81e853)

### Actual results

X-axis values are seemingly treated as categorical values:

![image](https://github.com/apache/superset/assets/1220356/2d3aa5b5-9345-4da1-9390-44d8e02c0304)

Any distribution/scatter/histogram plots based on this functionality are actually unusable due to this bug.

### Environment

(please complete the following information):

- browser type and version:
- superset version: 3.0.2rc1
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by michael-s-molina** [comment]
@villebro @zhaoyongjie Could you take a look at this one?

**Comment by villebro** [comment]
I will take a look, thanks for reporting @rumbin

**Comment by WojtekWaga** [comment]
Hey @villebro I checked the backend response and it's identical in this respect. I fed the json from Superset 2.0.1 backend into Superset 3.0.2rc1 frontend and it also misinterprets XAxis type. This made me convinced it's a FE issue. It looks like the getAxisTime defaults to AxisType.category in this case whereas it should return AxisType.time.

**Comment by villebro** [comment]
I was able to repro, investigating now

**Comment by rumbin** [comment]
@villebro, we found a fix on our side. @WojtekWaga will report here tomorrow and/or provide a PR.

**Comment by villebro** [comment]
@rumbin @WojtekWaga oops, I already opened a fix 🙁 Do you mind reviewing the PR #26087 to see if it's in line with the changes you were proposing?

**Comment by michael-s-molina** [comment]
Thank you all!
